templates/cluster-template.yaml

```diff
@@ -243,6 +243,19 @@ spec:
         kubectl --kubeconfig /etc/kubernetes/kubelet.conf
         patch node $(hostname)
         --type strategic -p '{"spec": {"providerID": "ionos://'$${system_uuid}'"}}'
+      - rm /etc/ssh/ssh_host_*
+      - ssh-keygen -t rsa -b 4096 -f /etc/ssh/ssh_host_rsa_key -N ""
+      - ssh-keygen -t ed25519 -f /etc/ssh/ssh_host_ed25519_key -N ""
+      - sed -i 's/^\#HostKey \/etc\/ssh\/ssh_host_\(rsa\|ed25519\)_key$/HostKey \/etc\/ssh\/ssh_host_\1_key/g' /etc/ssh/sshd_config
+      - awk '$5 >= 3071' /etc/ssh/moduli > /etc/ssh/moduli.safe
+      - mv /etc/ssh/moduli.safe /etc/ssh/moduli
+      - iptables -I INPUT -p tcp --dport 22 -m state --state NEW -m recent --set
+      - iptables -I INPUT -p tcp --dport 22 -m state --state NEW -m recent --update --seconds 10 --hitcount 10 -j DROP
+      - ip6tables -I INPUT -p tcp --dport 22 -m state --state NEW -m recent --set
+      - ip6tables -I INPUT -p tcp --dport 22 -m state --state NEW -m recent --update --seconds 10 --hitcount 10 -j DROP
+      - DEBIAN_FRONTEND=noninteractive apt install -q -y netfilter-persistent iptables-persistent
```

Comment:
These are not persisted, so they are gone after reboot. Pls test those things.

From `/var/log/cloud-init-output.log`:
```
[2024-06-12 09:52:59] WARNING: apt does not have a stable CLI interface. Use with caution in scripts.
[2024-06-12 09:52:59]
[2024-06-12 09:52:59] Reading package lists...
[2024-06-12 09:52:59] Building dependency tree...
[2024-06-12 09:52:59] Reading state information...
[2024-06-12 09:53:00] E: Unable to locate package netfilter-persistent
[2024-06-12 09:53:00] E: Unable to locate package iptables-persistent
[2024-06-12 09:53:00] netfilter-persistent: unrecognized service
```

What's missing is `apt update`, or `apt-get update`, as we're in a script, as the warning complains.
However, I prefer https://cloudinit.readthedocs.io/en/latest/reference/examples.html#update-apt-database-on-first-boot and https://cloudinit.readthedocs.io/en/latest/reference/examples.html#install-arbitrary-packages for cloud-init instead of bloating up `postKubeadmCommands` even more.

Reply:
For the paper trail: I mixed up cloud-init and the similar but limited capabilites offered e.g. by `KubeadmConfigTemplate`. The latter has none of the linked cloud-init options, so we continue with `postKubeadmCommands`.